lib/core/proxy.c

```diff
@@ -808,6 +813,8 @@ static void on_generator_dispose(void *_self)
         h2o_buffer_dispose(&self->last_content_before_send);
     }
     h2o_doublebuffer_dispose(&self->sending);
+    if (self->generator_disposed)
```

Comment:
```suggestion
    if (self->generator_disposed != NULL)
```
Let's use boolean as the result of an if expression (https://github.com/h2o/h2o/wiki/Coding-Style#if).